packages/x-date-pickers/src/tests/describeGregorianAdapter/testLocalization.ts

```diff
@@ -1,14 +1,51 @@
 import { expect } from 'chai';
+import sinon from 'sinon';
+import { AdapterFormats } from '@mui/x-date-pickers/models';
+import { cleanText } from 'test/utils/pickers-utils';
 import { DescribeGregorianAdapterTestSuite } from './describeGregorianAdapter.types';
+import { TEST_DATE_ISO_STRING } from './describeGregorianAdapter.utils';
 
 export const testLocalization: DescribeGregorianAdapterTestSuite = ({ adapter }) => {
+  const testDateIso = adapter.date(TEST_DATE_ISO_STRING)!;
+
   it('Method: formatNumber', () => {
     expect(adapter.formatNumber('1')).to.equal('1');
   });
 
   it('Method: getMeridiemText', () => {
     expect(adapter.getMeridiemText('am')).to.equal('AM');
     expect(adapter.getMeridiemText('pm')).to.equal('PM');
+
+    // Moment only translates for 12-hour cycle.
+    if (adapter.lib === 'moment') {
+      const stub = sinon.stub(adapter, 'is12HourCycleInCurrentLocale').returns(false);
+      expect(adapter.getMeridiemText('am')).to.equal('AM');
+      expect(adapter.getMeridiemText('pm')).to.equal('PM');
+      stub.restore();
+    }
```

Comment:
Isn't it now the case that only moment adapters are using this test? 🤔 
Shouldn't it have been something like this?
```ts
let stub;
if (adapter.lib === 'moment') {
  stub = sinon.stub(adapter, 'is12HourCycleInCurrentLocale').returns(false);
}
expect(adapter.getMeridiemText('am')).to.equal('AM');
expect(adapter.getMeridiemText('pm')).to.equal('PM');
if (stub) {
  stub.restore();
}
```

Reply:
No because in order to have 100% test coverage, on moment we want to test both with and without 12 hours cycle.

Reply:
Ah, sorry, so, there is another test asserting the logic without a stub..?
In that case, yeah, all is great! 👌 ✅